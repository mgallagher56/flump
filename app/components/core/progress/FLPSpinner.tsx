import React from 'react';

import type { SpinnerProps } from '@chakra-ui/react';
import { Spinner } from '@chakra-ui/react';

const FLPSpinner: React.FC<SpinnerProps> = (props) => {
  return <Spinner {...props} />;
};

export default FLPSpinner;
