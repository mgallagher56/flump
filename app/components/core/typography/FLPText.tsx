import React from 'react';

import type { TextProps } from '@chakra-ui/react';
import { Text } from '@chakra-ui/react';

const FLPText: React.FC<TextProps> = ({
  fontSize = 'md',
  fontWeight = 'normal',
  lineHeight = 'base',
  letterSpacing = 'normal',
  ...props
}) => {
  return (
    <Text fontSize={fontSize} fontWeight={fontWeight} lineHeight={lineHeight} letterSpacing={letterSpacing} {...props}>
      {props.children}
    </Text>
  );
};

export default FLPText;
