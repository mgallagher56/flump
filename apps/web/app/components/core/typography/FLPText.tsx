import type { FC } from 'react';

import type { TextProps } from '@chakra-ui/react';
import { Text } from '@chakra-ui/react';

const FLPText: FC<TextProps> = ({
  fontSize = 'md',
  fontWeight = 'normal',
  lineHeight = 'base',
  letterSpacing = 'normal',
  ...props
}) => {
  return (
    <Text fontSize={fontSize} fontWeight={fontWeight} letterSpacing={letterSpacing} lineHeight={lineHeight} {...props}>
      {props.children}
    </Text>
  );
};

export default FLPText;
