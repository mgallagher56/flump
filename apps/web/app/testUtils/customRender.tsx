// create a custom render function with ChakraProvider
import { type ReactNode } from 'react';

import { type RenderOptions, type RenderResult, render as rtlRender } from '@testing-library/react';
import userEvent, { type UserEvent } from '@testing-library/user-event';
import { Provider } from '~/components/ui/provider';

const user = userEvent.setup();

const customRender = (ui: ReactNode, options?: RenderOptions): RenderResult & { user: UserEvent } => {
  const wrappedUi = <Provider>{ui}</Provider>;

  return {
    ...rtlRender(wrappedUi, options),
    user
  };
};

export default customRender;
