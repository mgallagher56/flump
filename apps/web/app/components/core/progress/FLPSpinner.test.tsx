import customRender from '~/testUtils/customRender';

import FLPText from '~/components/core/typography/FLPText';

import FLPSpinner from './FLPSpinner';

describe('FLPSpinner', () => {
  test('renders as expected', () => {
    const { container } =customRender(<FLPSpinner />);
    expect(container).toMatchSnapshot();
  });
  test('renders as expected with a size', () => {
    const { container } =customRender(<FLPSpinner size="xl" />);
    expect(container).toMatchSnapshot();
  });
  test('renders as expected with a label', () => {
    const { container } =customRender(
      <FLPSpinner>
        <FLPText>Processing...</FLPText>
      </FLPSpinner>
    );
    expect(container).toMatchSnapshot();
  });
});
