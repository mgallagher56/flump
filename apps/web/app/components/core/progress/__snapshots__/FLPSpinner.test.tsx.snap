// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FLPSpinner > renders as expected 1`] = `
<div>
  <script
    nonce=""
  >
    ((e,r,s,u,d,m,l,h)=&gt;
    {let c=document.documentElement,v=["light","dark"];function p(i){(Array.isArray(e)?e:[e]).forEach(y=&gt;
    {let k=y==="class",S=k&&m?d.map(f=&gt;
    m[f]||f):d;k?(c.classList.remove(...S),c.classList.add(i)):c.setAttribute(y,i)}),R(i)}function R(i){h&&v.includes(i)&&(c.style.colorScheme=i)}function a(){return window.matchMedia("(prefers-color-scheme: dark)").matches?"dark":"light"}if(u)p(u);else try{let i=localStorage.getItem(r)||s,y=l&&i==="system"?a():i;p(y)}catch(i){}})("class","theme","system",null,["light","dark"],null,true,true)
  </script>
  <span
    class="chakra-spinner css-11t4vt7"
  />
</div>
`;

exports[`FLPSpinner > renders as expected with a label 1`] = `
<div>
  <script
    nonce=""
  >
    ((e,r,s,u,d,m,l,h)=&gt;
    {let c=document.documentElement,v=["light","dark"];function p(i){(Array.isArray(e)?e:[e]).forEach(y=&gt;
    {let k=y==="class",S=k&&m?d.map(f=&gt;
    m[f]||f):d;k?(c.classList.remove(...S),c.classList.add(i)):c.setAttribute(y,i)}),R(i)}function R(i){h&&v.includes(i)&&(c.style.colorScheme=i)}function a(){return window.matchMedia("(prefers-color-scheme: dark)").matches?"dark":"light"}if(u)p(u);else try{let i=localStorage.getItem(r)||s,y=l&&i==="system"?a():i;p(y)}catch(i){}})("class","theme","system",null,["light","dark"],null,true,true)
  </script>
  <span
    class="chakra-spinner css-11t4vt7"
  >
    <p
      class="css-gyi3bv"
    >
      Processing...
    </p>
  </span>
</div>
`;

exports[`FLPSpinner > renders as expected with a size 1`] = `
<div>
  <script
    nonce=""
  >
    ((e,r,s,u,d,m,l,h)=&gt;
    {let c=document.documentElement,v=["light","dark"];function p(i){(Array.isArray(e)?e:[e]).forEach(y=&gt;
    {let k=y==="class",S=k&&m?d.map(f=&gt;
    m[f]||f):d;k?(c.classList.remove(...S),c.classList.add(i)):c.setAttribute(y,i)}),R(i)}function R(i){h&&v.includes(i)&&(c.style.colorScheme=i)}function a(){return window.matchMedia("(prefers-color-scheme: dark)").matches?"dark":"light"}if(u)p(u);else try{let i=localStorage.getItem(r)||s,y=l&&i==="system"?a():i;p(y)}catch(i){}})("class","theme","system",null,["light","dark"],null,true,true)
  </script>
  <span
    class="chakra-spinner css-1x6hz8j"
  />
</div>
`;
